src/com/jaquadro/minecraft/storagedrawers/client/renderer/TileEntityDrawersRenderer.java

```diff
@@ -259,11 +259,39 @@ private void renderFastItem (ChamRender renderer, ItemStack itemStack, TileEntit
             renderHandlers.get(i).render(tile, tile, slot, 0, partialTickTime);
         }
 
-        GlStateManager.disableLighting();
+        // At the time GL_LIGHT* are configured, the coordinates are transformed by the modelview
+        // matrix. The transformations used in `RenderHelper.enableGUIStandardItemLighting` are
+        // suitable for the orthographic projection used by GUI windows, but they are just a little
+        // bit off when rendering a block in 3D and squishing it flat. An additional term is added
+        // to account for slightly different shading on the half-size "icons" in 1x2 and 2x2
+        // drawers due to the extreme angles caused by flattening the block (as noted below).
 
+        GlStateManager.pushMatrix();
+        GlStateManager.rotate(-30F, 0.0F, 1.0F, 0.0F);
+        GlStateManager.rotate(190F - size * 12F, 1.0F, 0.0F, 0.0F);
+        RenderHelper.enableStandardItemLighting();
+        GlStateManager.popMatrix();
+
+        // GL_POLYGON_OFFSET is used to offset flat icons toward the viewer (-Z) in screen space,
+        // so they always appear on top of the drawer's front space.
         GlStateManager.enablePolygonOffset();
         GlStateManager.doPolygonOffset(-1, -1);
 
+        // DIRTY HACK: Fool GlStateManager into thinking GL_RESCALE_NORMAL is enabled, but disable
+        // it using popAttrib This prevents RenderItem from enabling it again.
+        //
+        // Normals are transformed by the inverse of the modelview and projection matrices that
+        // excludes the translate terms. When put through the extreme Z scale used to flatten the
+        // block, this makes them point away from the drawer face at a very sharp angle. These
+        // normals are no longer unit scale (normalized), and normalizing them via
+        // GL_RESCALE_NORMAL causes a loss of precision that results in the normals pointing
+        // directly away from the face, which is visible as the block faces having identical
+        // (dark) shading.
+
+        GlStateManager.pushAttrib();
+        GlStateManager.enableRescaleNormal();
```

Comment:
Should we ask GlStateManager to disable rescale normal when we're done, so that it's not left in a lying state?